client/driver/ConfParser.py

```diff
@@ -0,0 +1,69 @@
+#
+# OtterTune - ConfParser.py
+#
+# Copyright (c) 2017-18, Carnegie Mellon University Database Group
+#
+'''
+Created on Mar 23, 2018
+@author: Jacky, bohan, Dongsheng
+'''
+
+import sys
+import json
+from collections import OrderedDict
+
+
+def change_postgres_conf(recommendation, postgresqlconf):
+    lines = postgresqlconf.readlines()
+    settings_idx = lines.index("# Add settings for extensions here\n")
+    postgresqlconf.seek(0)
+    postgresqlconf.truncate(0)
+
+    lines = lines[0:(settings_idx + 1)]
+    for line in lines:
+        postgresqlconf.write(line)
+
+    for (knob_name, knob_value) in list(recommendation.items()):
+        postgresqlconf.write(str(knob_name) + " = " + str(knob_value) + "\n")
+
+
+def change_oracle_conf(recommendation, oracle_conf):
+    lines = oracle_conf.readlines()
+    signal = "# configurations recommended by ottertune:\n"
+    if signal not in lines:
+        oracle_conf.write('\n' + signal)
+        oracle_conf.flush()
+        oracle_conf.seek(0)
+        lines = oracle_conf.readlines()
+    settings_idx = lines.index(signal)
+
+    oracle_conf.seek(0)
+    oracle_conf.truncate(0)
+
+    lines = lines[0:(settings_idx + 1)]
+    for line in lines:
+        oracle_conf.write(line)
+
+    for (knob_name, knob_value) in list(recommendation.items()):
```

Comment:
nitpick: you doesn't need the parenthesis surrounding `knob_name` and `knob_value`. Also, you don't need to change it to a `list()`

Reply:
I would lines 47-68 as such:
```
out_lines = [ f"{k} = {v}" for k, v in recommendations.items()]
oracle_conf.write('\n'.join(out_lines))
```